components/search/src/filter.rs

```diff
@@ -217,4 +295,401 @@ mod tests {
             }
         )
     }
+
+    lazy_static! {
+        static ref ENGINES_LIST: Vec<SearchEngineDefinition> = vec![
```

Comment:
Using lazy_static seemed to be the simplest way of creating this as a global so that it could be used across multiple tests - `SearchEngineDefinition` properties all need to be `String`s rather than `str`s, so need to have dynamic allocation afaict.

@linabutler If there's a better way of doing this, I'm open to improvements.

Reply:
You're totally right; we can't avoid the dynamic allocation here. `lazy_static` is ["soft-deprecated"](https://github.com/rust-lang-nursery/lazy-static.rs/issues/214); the standardized replacement for it is [`std::sync::LazyLock`](https://doc.rust-lang.org/stable/std/sync/struct.LazyLock.html), but that's only stable in 1.80+.

In our newer components, we've been using [`once_cell`](https://docs.rs/once_cell/latest/once_cell). [`once_cell::sync::Lazy`](https://docs.rs/once_cell/latest/once_cell/sync/struct.Lazy.html) should do the trick; `std::sync::LazyLock` is based on it!

```rs
static ENGINES_LIST: Lazy<Vec<SearchEngineDefinition>> = Lazy::new(|| {
    vec![
        SearchEngineDefinition {
            // ...
        },
        SearchEngineDefinition
            // ...
        },
    ]
});
```